services/node_manager/src/main/java/com/futurewei/alcor/nodemanager/dao/NmmInfoRepository.java

```diff
@@ -0,0 +1,161 @@
+/*Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.nodemanager.dao;
+
+import com.futurewei.alcor.common.db.CacheException;
+import com.futurewei.alcor.common.db.CacheFactory;
+import com.futurewei.alcor.common.db.ICache;
+import com.futurewei.alcor.common.db.Transaction;
+import com.futurewei.alcor.common.db.repo.ICacheRepository;
+import com.futurewei.alcor.common.stats.DurationStatistics;
+import com.futurewei.alcor.web.entity.node.NmmInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.Map;
+
+@Repository
+public class NmmInfoRepository implements ICacheRepository<NmmInfo> {
+    private static final Logger logger = LoggerFactory.getLogger(NmmInfoRepository.class);
```

Comment:
@pkommoju Does this cache for ncminfo in node manager? If it is, it's better to name the class `NcmInfoRepository`.

Reply:
The name NmmInfoRepository may be confusing but NcmInfoRepository could
 be even more confusing because in this map, the value is NmmInfo and
 NmmInfo contains NcmInfo.
I think NmmInfoRepository is better because what we see is NmmInfo.

I think containg NcmInfo in NmmInfo is simpler than inheriting from. NmmInfo is an artificial construct. Ideally I would have liked Map<NcmId, Pair<NcmInfo, NodeInfo>> but that seemed complicated nesting exposed to outside.
Let me know if this is a reasonable compromise.

Reply:
@pkommoju  So, you want to create a cache to map from NcmId to <NcmInfo, NodeInfo> pair?
If so, why not just put a list of NodeIds into the NcmInfo entity? and make a map <NcmId, NcmInfo>.
In that way, you can query a NcmInfo and its NodeIds using a NcmId.
You can also put NcmId into NodeInfo entity so that you can check a certain Node's Mcminfo.

Reply:
If the map entry is like <ncmId, NcmInfo> where NcmInfo is {other ncm info,  list<nodeId>}, then adding/deleting nodes would get more expensive, doesn't it?
First, one map operation to read the map entry, next add/del nodeid entry, and then write back the map entry.
With the current scheme add and delete touch the cache only once and don't have to mainipulate a list/set of nodeids.